src/components/SingleBlock.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {useEffect, useState} from 'react';
+import './SingleBlock.css';
+import {
+    useParams
+} from "react-router-dom";
+import {Constants, fetchConstantsData, fetchSingleBlock} from "../helpers/api";
+import StatRow from "./SingleBlock/StatRow";
+import SingleBlockViewHeader from "./SingleBlock/SingleBlockViewHeader";
+import { connect } from 'react-redux';
+import ProgressBar from "./SingleBlock/ProgressBar";
+import ClusterGraph from "./Graphs/ClusterGraph";
+import {Block, BlocksEntity, Body, Header, Pow} from "../helpers/Blocks";
+import { Link } from "react-router-dom";
+
+interface Props {
+    block: BlocksEntity[];
+}
+
+function SingleBlock({ block }: Props) {
+
+    const { id } = useParams();
+
+    const [singleBlock, setSingleBlock] = useState({} as Block);
+    const [blockHeader, setblockHeader] = useState({} as Header);
+    const [blockPow, setblockPow] = useState({} as Pow);
+    const [blockBody, setblockBody] = useState({} as Body);
+    const [constants, setConstants] = useState({} as Constants);
+
+    useEffect(() => {
+        try {
+            id && fetchSingleBlock(id).then((block) => {
+                if(block && block.block) {
+                    setSingleBlock(block.block);
+                    setblockHeader(block.block.header as Header);
+                    setblockPow(block.block.header.pow as Pow);
+                    setblockBody(block.block.body as Body);
+                }
+            });
+        } catch (e) {
+            console.error(e);
+        }
+    }, [id]);
+
+    useEffect(() => {
+        try {
+            id && fetchConstantsData().then((constants: Constants) => {
+                setConstants(constants as Constants);
+            });
+        } catch (e) {
+            console.error(e);
+        }
+    }, [id]);
```

Comment:
Why are you calling this here again? The constants were already fetched in `store/index.ts` `fetchConstants` use redux to get the values.